presto-main/src/main/java/com/facebook/presto/operator/project/InputPageProjection.java

```diff
@@ -56,32 +57,44 @@ public InputChannels getInputChannels()
     }
 
     @Override
-    public PageProjectionOutput project(ConnectorSession session, DriverYieldSignal yieldSignal, Page page, SelectedPositions selectedPositions)
+    public Work<Block> project(ConnectorSession session, DriverYieldSignal yieldSignal, Page page, SelectedPositions selectedPositions)
     {
-        return new InputPageProjectionOutput(page, selectedPositions);
+        return new InputPageProjectionWork(page, selectedPositions);
     }
 
-    private class InputPageProjectionOutput
-            implements PageProjectionOutput
+    private class InputPageProjectionWork
+            implements Work<Block>
     {
         private final Block block;
         private final SelectedPositions selectedPositions;
+        private Block result;
 
-        public InputPageProjectionOutput(Page page, SelectedPositions selectedPositions)
+        public InputPageProjectionWork(Page page, SelectedPositions selectedPositions)
```

Comment:
This `Work` looks like a constant work. Consider using that instead?

(If this is temporary and should become a real work at some point, feel free to leave it as is. But please add a comment.)